### PR TITLE
Integrate GitHub app test plan

### PR DESCRIPTION
**Pull request by parthfloyd**
- [ ] Tested on local & docker
- [ ] Documentation Updated
- [ ] Attached required context / screenshot

### CODE REVIEW

**Comment by parthfloyd** [comment]
To update in the docs:
1. For Github Bot in `repository permission` add: `read & write` permission for `Issues`  & subscribe to `issue comment` event.
2. Added env variable for github bot name to support mention feature.

File: server/knowledge_graph/flow.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 import hashlib
 import psycopg2
 from server.utils.github_helper import GithubService
-from utils.graph_db_helper import Neo4jGraph
+from server.utils.graph_db_helper import Neo4jGraph
```
Comment (dhirenmathur) [inline server/knowledge_graph/flow.py:11]: good catch!

File: server/utils/github_helper.py [hunk 2]
```diff
@@ -75,4 +75,20 @@ def fetch_method_from_repo(node):
...
+            "body":comment
+        }
+        url = f"https://api.github.com/repos/{owner}/{repo}/issues/{issue_number}/comments"
+        headers = {
+            "Accept": "application/vnd.github+json",
+            "Authorization": f"Bearer {installation_auth.token}",
+            "X-GitHub-Api-Version": "2022-11-28"
+        }
+
+        return requests.post(url, headers=headers, json=data)
```
Comment (dhirenmathur) [inline server/utils/github_helper.py:94]: any reason to not use pygithub and call the API directly?

File: server/routers/webhook.py [hunk 3]
```diff
@@ -122,3 +130,190 @@ async def parse_repos(payload, request: Request):
...
+        repo = repo_name.split('/')[1]
+
+        url = f"https://api.github.com/repos/{owner}/{repo}/pulls/{pull_number}"
+        headers = {
+            "Accept": "application/vnd.github+json",
+            "Authorization": f"Bearer {installation_auth.token}",
+            "X-GitHub-Api-Version": "2022-11-28"
+        }
+
+        return requests.get(url, headers=headers)
```
Comment (dhirenmathur) [inline server/routers/webhook.py:319]: any reason to not use pygithub and call the API directly?

File: server/routers/webhook.py [hunk 4]
```diff
@@ -122,3 +130,190 @@ async def parse_repos(payload, request: Request):
...
+    repo_name = payload['repository']['full_name']
+    branch_name = payload['pull_request']['head']['ref']
+    base_branch_name = payload['pull_request']['base']['ref']
+    owner = payload["pull_request"]["head"]["repo"]["owner"]["login"]
+    pr_number = payload["pull_request"]["number"]
+
+    #creating github object & fetching repo details
+    installation_auth = get_installation_auth(payload)
+    github = Github(auth=installation_auth)
+    repo_details = github.get_repo(repo_name) #TODO: parse this from payload
```
Comment (dhirenmathur) [inline server/routers/webhook.py:195]: can remove the todo comment

File: server/utils/github_helper.py (hunk 2 above)
Comment (parthfloyd) [reply server/utils/github_helper.py:94]: Great suggestion, I'll use it wherever applicable now onwards.

**Review by dhirenmathur — APPROVED** [review]

**Review by dhirenmathur — APPROVED** [review]

**Pull request merged**